pkg/hubble/relay/observer/server.go

```diff
@@ -235,3 +240,27 @@ func (s *Server) ServerStatus(ctx context.Context, req *observerpb.ServerStatusR
 	}
 	return resp, g.Wait()
 }
+
+// filterNodes applies req's nodes filters to peers and returns peers. peers is
```

Comment:
My initial read of this confused me, as it sounds a bit contradictory. The final sentence should mention that the backing array is re-used on top of returning a new slice.

Reply:
Thanks, I've updated the function's comment to hopefully be clearer.